### DISCUSSION TITLE
ASR Transducer Wav2vec on TIMIT Database --issue--   forward() takes 2 positional arguments but 3 were given

### DISCUSSION THREAD

**Discussion by smzphd** [speechbrain/speechbrain#1856] [Q&A] [answered]
Hi,

I am new to speechbrain and trying to get some recipes running, specially   ASR Transducer Wavevec on TIMIT Database 
and during that I am facing this issue >>   forward() takes 2 positional arguments but 3 were given.

I know discussions on similar errors are being done on the forum before but no definite solution is presented or may be i am too novice to understand. 

I am using local TIMIT Dataset and following the recipe 

python train_wav2vec.py hparams/train_wav2vec.yaml --data_folder ./data/TIMIT/TIMIT/

Some weights of the model checkpoint at facebook/wav2vec2-large-lv60 were not used when initializing Wav2Vec2Model: ['project_hid.weight', 'quantizer.weight_proj.bias', 'project_q.weight', 'project_hid.bias', 'project_q.bias', 'quantizer.weight_proj.weight', 'quantizer.codevectors']
- This IS expected if you are initializing Wav2Vec2Model from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing Wav2Vec2Model from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
speechbrain.core - Beginning experiment!
speechbrain.core - Experiment folder: results/augment_wav2vec/1234
timit_prepare - Skipping preparation, completed in previous run.
speechbrain.dataio.encoder - Load called, but CTCTextEncoder is not empty. Loaded data will overwrite everything. This is normal if there is e.g. an unk label defined at init.
speechbrain.core - Info: jit_module_keys arg from hparam file is used
speechbrain.core - Info: auto_mix_prec arg from hparam file is used
speechbrain.core - 315.6M trainable parameters in ASR_Brain
speechbrain.utils.checkpoints - Would load a checkpoint here, but none found yet.
speechbrain.utils.epoch_loop - Going into epoch 1
  0%|                                                                                           | 0/462 [00:00<?, ?it/s]
speechbrain.core - Exception:
Traceback (most recent call last):
  File "train_wav2vec.py", line 359, in <module>
    asr_brain.fit(
  File "/home/smz/env_38/lib/python3.8/site-packages/speechbrain/core.py", line 1153, in fit
    self._fit_train(train_set=train_set, epoch=epoch, enable=enable)
  File "/home/smz/env_38/lib/python3.8/site-packages/speechbrain/core.py", line 1009, in _fit_train
    loss = self.fit_batch(batch)
  File "train_wav2vec.py", line 176, in fit_batch
    outputs = self.compute_forward(batch, sb.Stage.TRAIN)
  File "train_wav2vec.py", line 39, in compute_forward
    feats = self.modules.wav2vec2(wavs, wav_lens)
  File "/home/smz/env_38/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
TypeError: forward() takes 2 positional arguments but 3 were given

My system has 
speechbrain              0.5.13
huggingface-hub-0.12.0
torch                    1.13.1
torchaudio               0.13.1
torchsummary             1.5.1
torchvision              0.14.1

Please suggest the solution.

**Comment by TParcollet** [comment]
Hi, how did you install SpeechBrain? Pip install SpeechBrain or git clone?

**Reply by smzphd** [reply]
pip install speechbrain
and copy recipe folder

**Comment by anautsch** [comment, marked as answer]
Hi @smzphd how did you install SpeechBrain?

The code line of your concern was changed recently:
https://github.com/speechbrain/speechbrain/blame/develop/recipes/TIMIT/ASR/seq2seq/train_with_wav2vec2.py#L40

This change will be available via `pip install speechbrain` within the next days, when we release v0.5.14.

Until then, please install from our git `develop` branch, as of the general README:
```bash
# You have this already, right?
git clone https://github.com/speechbrain/speechbrain.git
cd speechbrain

# Then, you'll need only these steps
pip install -r requirements.txt
pip install --editable .
```

It's an intuition on my side, please lmk if this helps you, or not.

**Reply by smzphd** [reply]
ok Thanx for swift response.

**Reply by smzphd** [reply]
@anautsch Thanx 
You were write .
It worked fine.

**Comment by Elquintas** [comment]
Greetings, 

I'm facing the same issue on the commonvoice recipes (CTC + wav2vec2 and also transformer + wav2vec2). i've installed speechbrain recently (~2 weeks ago, version 5.13).

Curiously, the system runs locally without the presented issue, however when I try to train it remotely on a cluster it displays this error. At first I thought it would be related to python dependencies, however other CV recipes worked fine (transducer, seq2seq without wav2vec, etc) using the same singularity.

I've seen the proposed changes in the TIMIT recipe, however in my train_with_wav2vec.py I already have the line "feats = self.modules.wav2vec2(wavs, wav_lens)" changed.

Do you have any intuition behind this error on this particular recipe?

Thank you very much in advance,
Best regards.

**Comment by anautsch** [comment]
Greetings @Elquintas,

if SB is on your cluster installed with `pip install speechbrain`, please use the local install with git clone & from the develop branch as shown above. We are working on releasing an updated minor version soon.

**Reply by Elquintas** [reply]
Thank you for the fast response @anautsch , 
will try and keep you updated.
